app/models/bitflyer_connection.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class BitflyerConnection < ApplicationRecord
+  SUPPORTED_CURRENCIES = ["BAT", "USD", "BTC", "ETH"].freeze
+
+  belongs_to :publisher
+
+  validates :recipient_id, uniqueness: true, allow_blank: true
+
+  attr_encrypted :access_token, :refresh_token, key: :encryption_key
+
+  after_save :update_default_currency, if: -> { saved_change_to_default_currency? }
+
+  validates :default_currency, inclusion: { in: SUPPORTED_CURRENCIES }, allow_nil: true
+
+  after_destroy :selected_wallet_provider
```

Comment:
Hmm, not completely certain but I think we want to use a `before_destroy` (I don't know if calling `.destroy` will wipe the `id` field)